### DISCUSSION TITLE
Issue when dealing with saveCall

### DISCUSSION THREAD

**Discussion by w-mazed** [ionic-team/capacitor#5704] [Q&A] [answered]
Hi,

I need to resolve my PluginCall from another function, so I saved it through Bridge, and resolve it later, but I've message this message :
```
Attempt to invoke virtual method 'void com.getcapacitor.PluginCall.resolve(com.getcapacitor.JSObject)' on a null object reference
```

So this is a bit of my code, can you tell me guys where I'm wrong :
```
@CapacitorPlugin(name = "ChainwayR6")
public class ChainwayR6Plugin extends Plugin {

    Bridge bridge;
    String callbackId;
    PluginCall savedCall;

    @PluginMethod
    public void connect(PluginCall call) throws JSONException {
        String address = call.getString("address");

        implementation.connect(address);

        callbackId = call.getCallbackId();

        bridge.saveCall(call);

        // call.resolve(response);
    }


    @Subscribe
    public void onMessageEvent(MessageEvent event) {

        ChainwayR6Device deviceReader = event.getDeviceReader();

        JSObject response = new JSObject();

        JSObject props = new JSObject();
        props.put("success", event.getSuccess());

        response.put("response", props);

        savedCall = bridge.getSavedCall(callbackId);
        savedCall.resolve(response);

        bridge.releaseCall(callbackId);
        callbackId = null;
    }
```

Thanks.

**Comment by mlynch** [comment, marked as answer]
There's no guarantee that `getSavedCall` will return a non-null `PluginCall` so you should make sure it's not-null before using it.

Take a look at how the Geolocation plugin saves calls and reuses them: 

Retaining:
https://github.com/ionic-team/capacitor-plugins/blob/7563040983ad397e28616246e7ed5ffce69727c2/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/GeolocationPlugin.java#L94

Storing: 
https://github.com/ionic-team/capacitor-plugins/blob/7563040983ad397e28616246e7ed5ffce69727c2/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/GeolocationPlugin.java#L165

Using:
https://github.com/ionic-team/capacitor-plugins/blob/7563040983ad397e28616246e7ed5ffce69727c2/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/GeolocationPlugin.java#L155

Releasing:
https://github.com/ionic-team/capacitor-plugins/blob/7563040983ad397e28616246e7ed5ffce69727c2/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/GeolocationPlugin.java#L179

**Reply by w-mazed** [reply]
Thanks I'll take a look.
